zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java

```diff
@@ -78,43 +94,58 @@ public void flush(final BatchRequest batchRequest) throws PersistenceException {
         handler.flush(entity, batchRequest);
       }
     }
-    batchRequest.execute(customErrorHandlers);
+
+    batchRequest.execute(customErrorHandler);
+    observeRecordTimestamps();
     reset();
   }
 
-  public void reset() {
-    cachedEntities.clear();
+  private void observeRecordTimestamps() {
+    final var timestamps = new ArrayList<>(cachedRecordTimestamps.values());
+    cachedRecordTimestamps.clear();
+    metrics.observeRecordExportLatencies(timestamps);
   }
 
   public int getBatchSize() {
     return cachedEntities.size();
   }
 
-  public static class Builder {
-    private ExporterBatchWriter writer;
+  private void reset() {
+    cachedEntities.clear();
+  }
+
+  public static final class Builder {
+    private final CamundaExporterMetrics metrics;
+    private final Map<ValueType, List<ExportHandler>> handlers = new HashMap<>();
+    private BiConsumer<String, Error> customErrorHandler = (ignored, error) -> {};
 
-    public static Builder begin() {
-      final Builder builder = new Builder();
-      builder.writer = new ExporterBatchWriter();
-      return builder;
+    private Builder(final CamundaExporterMetrics metrics) {
+      this.metrics = metrics;
+    }
+
+    public static Builder begin(final CamundaExporterMetrics metrics) {
+      return new Builder(metrics);
+    }
+
+    @VisibleForTesting
+    static Builder begin() {
```

Comment:
:thought_balloon: Not super happy with this, but otherwise it means for test you always have to provide some default metrics implementation...